discovery/trends/v0.1/trends.md

```diff
@@ -0,0 +1,355 @@
+links. This "normalization" is not part of any standard and different
+fediverse software can have different and even incompatible
+implementations of this.
+
+That is why when sharing trending hashtags and links with fediverse
+servers, FASP do not need to take special care how to represent them and
+can freely choose the representation (e.g. the first one they
+encountered or a normalized version). Fediverse servers SHOULD handle
+them just like they handle these representations in other contexts.
+
+When computing trends however, FASP SHOULD consider normalization to
+minimize the risk of obvious duplicates.
+
+For hashtags possible strategies for this besides case-insensitive
+handling of names include "ASCII folding" and using existing character
+transliteration rules. As all of these approaches have their downsides
+and can lead to two different words being erroneously treated as the
+same it is up to FASP to decide which strategy to use.
+
+For links [RFC-3986](https://tools.ietf.org/html/rfc3986.html) describes
+different normalization techniques in section 6.
+
+## Availability of Content for Hashtags and Links
+
+When a fediverse server requests trending hashtags or links it will in
+most cases want to display them to users and have a way for users to see
+how and where those have been used. Since FASP have a wider view of the
+fediverse it might be the case that the fediverse server has no content
+at all that references a trending hashtag or link. Or it might not have
+recent content that does so.
+
+To help fediverse servers in this situation discovery FASP include a
+couple of example URIs of content with every result of a query for
+hashtags and links trends. This allows fediverse servers to quickly fill
+their local caches with some content for the received trends. See the
+sections "Requesting Trending Hashtags" and "Requesting Trending Links"
+below for details.
+
+## Requesting Trends
+
+### Common Request Options
+
+All requests for trends from a fediverse server to FASP are HTTP `GET`
+requests that allow - with one exception - using the same optional
+parameters:
+
+* `withinLastHours`: This MUST be a positive integer specifying the
+  number of hours up to the current time for which to compute trends.
+  The minimum value of `1` means records trending within the last hour,
+  while the allowed maximum of `336` means within the last fortnight.
+  If omitted defaults to `24`.
+* `maxCount`: This MUST be a positive integer specifying the maximum
+  number of results that should be returned. If omitted defaults to
+  `20`.
+* `language`: An ISO-639 language code to only receive results in or
+  relevant for the specific language. If omitted results can be in or
+  relevant for any language.
+
+### Common Response Attributes 
+
+All responses include a list of results. Individual result objects
+share the following common keys and values:
+
+* `rank`: A positive integer less than or equal `100` representing the
```

Comment:
A *relative rank* will make it harder to compare trends over time.

Reply:
Yes! Which is intentional. This is API is only about _current_ trends. Show the user a couple of things are relevant _right now_.

I know that one can easily construct other use cases. Being able to query and compare historic trends for example is probably super interesting, especially for researchers. However, this does not fall under the umbrella of "search and discovery", so is totally out-of-scope here.

Reply:
Even with *current* trends, relative ranks are useless if anyone wants to combine the data from multiple FASPs.

Reply:
Not necessarily. I addressed this in this comment here: https://github.com/mastodon/fediverse_auxiliary_service_provider_specifications/pull/45#issuecomment-2650353942